lib/spack/spack/cmd/env.py

```diff
@@ -118,19 +209,40 @@ def env_activate_setup_parser(subparser):
         help="decorate the command line prompt when activating",
     )
 
-    env_options = subparser.add_mutually_exclusive_group()
-    env_options.add_argument(
+    subparser.add_argument(
         "--temp",
         action="store_true",
         default=False,
         help="create and activate an environment in a temporary directory",
     )
-    env_options.add_argument(
-        "-d", "--dir", default=None, help="activate the environment in this directory"
+    subparser.add_argument(
+        "--create",
+        action="store_true",
+        default=False,
+        help="if the environment doesn't exist, create it before activating",
```

Comment:
Nit:  it is more consistent with the help of similar options if this starts with a verb (e.g., `--temp`), such as:

```suggestion
        help="create and activate the environment if it doesn't exist",